src/function/aggregate/holistic/quantile.cpp

```diff
@@ -157,43 +196,59 @@ struct DiscreteQuantileOperation : public QuantileOperation<INPUT_TYPE> {
 	}
 };
 
-AggregateFunction GetDiscreteQuantileAggregateFunction(PhysicalType type) {
-	switch (type) {
-	case PhysicalType::INT16:
-		return AggregateFunction::UnaryAggregateDestructor<QuantileState, int16_t, int16_t,
-		                                                   DiscreteQuantileOperation<int16_t>>(LogicalType::SMALLINT,
-		                                                                                       LogicalType::SMALLINT);
-
-	case PhysicalType::INT32:
-		return AggregateFunction::UnaryAggregateDestructor<QuantileState, int32_t, int32_t,
-		                                                   DiscreteQuantileOperation<int32_t>>(LogicalType::INTEGER,
-		                                                                                       LogicalType::INTEGER);
-
-	case PhysicalType::INT64:
-		return AggregateFunction::UnaryAggregateDestructor<QuantileState, int64_t, int64_t,
-		                                                   DiscreteQuantileOperation<int64_t>>(LogicalType::BIGINT,
-		                                                                                       LogicalType::BIGINT);
-
-	case PhysicalType::INT128:
-		return AggregateFunction::UnaryAggregateDestructor<QuantileState, hugeint_t, hugeint_t,
-		                                                   DiscreteQuantileOperation<hugeint_t>>(LogicalType::HUGEINT,
-		                                                                                         LogicalType::HUGEINT);
-	case PhysicalType::FLOAT:
-		return AggregateFunction::UnaryAggregateDestructor<QuantileState, float, float,
-		                                                   DiscreteQuantileOperation<float>>(LogicalType::FLOAT,
-		                                                                                     LogicalType::FLOAT);
-
-	case PhysicalType::DOUBLE:
-		return AggregateFunction::UnaryAggregateDestructor<QuantileState, double, double,
-		                                                   DiscreteQuantileOperation<double>>(LogicalType::DOUBLE,
-		                                                                                      LogicalType::DOUBLE);
+template <typename INPUT_TYPE>
+AggregateFunction GetTypedDiscreteQuantileAggregateFunction(const LogicalType &type) {
+	return AggregateFunction::UnaryAggregateDestructor<QuantileState, INPUT_TYPE, INPUT_TYPE,
+	                                                   DiscreteQuantileOperation<INPUT_TYPE>>(type, type);
+}
+
+AggregateFunction GetDiscreteQuantileAggregateFunction(const LogicalType &type) {
+	switch (type.id()) {
```

Comment:
Could the discrete quantile use the physical type instead of the logical type to avoid several branches here?